### DISCUSSION TITLE
Windows Package Manager 1.7.10582

### DISCUSSION THREAD

**Discussion by ryfu-msft** [microsoft/winget-cli#4207] [Releases]
This release is the first stable release of Windows Package Manager 1.7 for Windows 10 (1809+), and Windows 11. Experimental features have been disabled in this release.

## Features
- The `winget repair` command is now available and can be used to repair a package that is having issues.
- Winget can enable Windows Features as a package dependency. 
- Winget now supports rebooting your machine if the installer return code indicates that a reboot is required. You can do this by passing the `--allow-reboot` flag.
- WinGet configuration can accept a URL rather than only a local file.
- Performance improvements to be more network efficient with the CDN along with better flows to handle scenarios with elevation requirements.
- Support for Entra Id authentication (formerly Azure Active Directory) for private REST sources. Note: The REST source reference implementation still needs to be updated.

## What's Changed
* Remove Invoke-CommandInDesktopPackage use by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/3658
* Improve packaged source updating by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/3657
* Add a bit of randomness to the wait time after source update failure by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/3661
* Reduce the size of the index by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/3666
* Log Com invocation startup telemetry and delay auto update time when invoked from explorer by @yao-msft in https://github.com/microsoft/winget-cli/pull/3665
* Enable COM API access to correlate with the tracking database only by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/3670
* Fix localized strings output by @mdanish-kh in https://github.com/microsoft/winget-cli/pull/3673
* Improve the version header detection and logging by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/3680
* Revert 3670 by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/3700
* Enable COM API access to correlate with the tracking database only by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/3703
* Use correct caller name in Com startup telemetry event by @yao-msft in https://github.com/microsoft/winget-cli/pull/3711
* Platform source should not be blocked by policy check by @yao-msft in https://github.com/microsoft/winget-cli/pull/3725
* Update docs for upgrade command by @KK-Designs in https://github.com/microsoft/winget-cli/pull/3639
* Honor 429 Retry-After by @msftrubengu in https://github.com/microsoft/winget-cli/pull/3718
* Make manifest retrieval choice more dynamic by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/3738
* Allow --accept-source-agreements with local manifests by @Trenly in https://github.com/microsoft/winget-cli/pull/3573
* Invoke ShellExecute on dism.exe for enabling Windows Features by @ryfu-msft in https://github.com/microsoft/winget-cli/pull/3659
* Allow --include-unknown with install by @Trenly in https://github.com/microsoft/winget-cli/pull/3752
* Broadcast WM_SETTINGCHANGE on change of path by @Trenly in https://github.com/microsoft/winget-cli/pull/3751
* Add `resume` command and support saving the argument state. by @ryfu-msft in https://github.com/microsoft/winget-cli/pull/3508
* Allow higher versions to satisfy the VCLibs dependency in Repair by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/3763
* Use package version as potential last update timestamp by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/3759
* Add missing condition for using toolset v143 when building JsonCppLib for arm by @florelis in https://github.com/microsoft/winget-cli/pull/3773
* Move SQLite base code by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/3790
* Fix non-test hook code for Windows Feature by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/3789
* Skip dependency evaluation with `--skip-dependencies` by @mdanish-kh in https://github.com/microsoft/winget-cli/pull/3784
* Download and install Workflow patches for skip dependencies by @yao-msft in https://github.com/microsoft/winget-cli/pull/3794
* Add experimental feature for initiating reboot for single package installs by @ryfu-msft in https://github.com/microsoft/winget-cli/pull/3631
* Microsoft.WinGet.Client Any CPU by @msftrubengu in https://github.com/microsoft/winget-cli/pull/3622
* Inform user if a module requires elevation by @msftrubengu in https://github.com/microsoft/winget-cli/pull/3758
* Add missing manifest fields by @hackean-msft in https://github.com/microsoft/winget-cli/pull/3757
* Configuration Schema 0.3 by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/3779
* Extend Configuration COM API for export by @florelis in https://github.com/microsoft/winget-cli/pull/3787
* Fix binskim issues by @ryfu-msft in https://github.com/microsoft/winget-cli/pull/3815
* Performance improvements by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/3808
* Fix OOP config helper by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/3873
* Revert manifest schemas to json draft-07 by @ryfu-msft in https://github.com/microsoft/winget-cli/pull/3875
* Allow Microsoft.WinGet.Client to run in any PowerShell session running as system by @msftrubengu in https://github.com/microsoft/winget-cli/pull/3816
* Fix signaling the app shutdown event running as admin by @msftrubengu in https://github.com/microsoft/winget-cli/pull/3874
* Configuration schema property descriptions added by @alexravenna in https://github.com/microsoft/winget-cli/pull/3499
* The initial yaml schema manifest for version 1.7.0 by @Madhusudhan-MSFT in https://github.com/microsoft/winget-cli/pull/3876
* Move to latest cppwinrt package across all projects by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/3868
* Add suggested dictionaries for spellchecking and remove unneeded words by @florelis in https://github.com/microsoft/winget-cli/pull/3885
* Remove unneeded words from spellchecking by @florelis in https://github.com/microsoft/winget-cli/pull/3890
* Update docs for winget commands by @KK-Designs in https://github.com/microsoft/winget-cli/pull/3909
* Ignore deprecation warning by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/3905
* Fix pipeline build error by @florelis in https://github.com/microsoft/winget-cli/pull/3937
* Add support for double-clicking on .wingetdev files for applying configuration by @florelis in https://github.com/microsoft/winget-cli/pull/3860
* Add a script to bootstrap running Pester tests by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/3899
* Add WingetDSC E2E tests by @ryfu-msft in https://github.com/microsoft/winget-cli/pull/3939
* Improve repair by @msftrubengu in https://github.com/microsoft/winget-cli/pull/3886
* Enable cmdlets for Windows PowerShell by @msftrubengu in https://github.com/microsoft/winget-cli/pull/3951
* Register restart for resume by @ryfu-msft in https://github.com/microsoft/winget-cli/pull/3858
* Allow user settings to control logging channels by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/3955
* Add package id, name, and source to install/update/uninstall result for PowerShell cmdlet by @ryfu-msft in https://github.com/microsoft/winget-cli/pull/3954
* Update Store Certs by @yao-msft in https://github.com/microsoft/winget-cli/pull/3968
* Introduce strong and weak comparisons between installers by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/3956
* Include framework packages during installed packages enumeration by @yao-msft in https://github.com/microsoft/winget-cli/pull/3975
* Allow winget configure from https location and extend winget configure validate for winget resource units by @yao-msft in https://github.com/microsoft/winget-cli/pull/3833
* Update comment around MotW application by @florelis in https://github.com/microsoft/winget-cli/pull/3979
* Restrict on agreement text only by @yao-msft in https://github.com/microsoft/winget-cli/pull/3998
* Create similarissues.yml by @craigloewen-msft in https://github.com/microsoft/winget-cli/pull/4035
* Update similarIssues.yml to not show 'fail' when no similar issues found  by @craigloewen-msft in https://github.com/microsoft/winget-cli/pull/4042
* Similar issues workflow explicit permission by @msftrubengu in https://github.com/microsoft/winget-cli/pull/4040
* Update WinGetUtilInterop project by @msftrubengu in https://github.com/microsoft/winget-cli/pull/4045
* Repair switch support for V1.7 YAML manifest by @Madhusudhan-MSFT in https://github.com/microsoft/winget-cli/pull/4041
* Update dependency on System.Data.SqlClient by @florelis in https://github.com/microsoft/winget-cli/pull/4083
* Use std::variant in ManifestYamlPopulator by @msftrubengu in https://github.com/microsoft/winget-cli/pull/4081
* Show only agreement related info during install by @Trenly in https://github.com/microsoft/winget-cli/pull/3999
* Support group processing of configurations by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/4059
* Improve progress handling for group processor by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/4121
* Shadow Manifest by @msftrubengu in https://github.com/microsoft/winget-cli/pull/4104
* Remove debug flags to unblock utils nuget release (#4030) by @msftrubengu in https://github.com/microsoft/winget-cli/pull/4127
* Revert "Remove debug flags to unblock utils nuget release (#4030)" by @msftrubengu in https://github.com/microsoft/winget-cli/pull/4128
* Support Microsoft Entra Id authentication for rest source by @yao-msft in https://github.com/microsoft/winget-cli/pull/4123
* Retarget to netstandard2.1 (#4130) by @msftrubengu in https://github.com/microsoft/winget-cli/pull/4132
* Make windows feature and reboot features stable by @ryfu-msft in https://github.com/microsoft/winget-cli/pull/4137
* Update links in Roadmap by @Trenly in https://github.com/microsoft/winget-cli/pull/4142
* Refactor pinning evaluation by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/4151
* Winget client rest source parsing for 1.7 manifest by @yao-msft in https://github.com/microsoft/winget-cli/pull/4155
* Add 1.7 manifest fields to WingetUtilsInterop by @yao-msft in https://github.com/microsoft/winget-cli/pull/4157
* Update fuzzer and integrate with OneFuzz by @ryfu-msft in https://github.com/microsoft/winget-cli/pull/4135
* Add additional switches for Inno Setup based installers by @SpecterShell in https://github.com/microsoft/winget-cli/pull/3562
* Create #658 - WinGet Download.md by @RDMacLachlan in https://github.com/microsoft/winget-cli/pull/2953
* Log configuration input hash in configuration summary telemetry by @yao-msft in https://github.com/microsoft/winget-cli/pull/4173
* Implement Export-WinGetPackage powershell cmdlet for `winget download` by @ryfu-msft in https://github.com/microsoft/winget-cli/pull/3977
* winget repair cli implementation by @Madhusudhan-MSFT in https://github.com/microsoft/winget-cli/pull/4168
* Fix typo in '#658 - WinGet Download.md' Spec File by @og-mrk in https://github.com/microsoft/winget-cli/pull/4179
* Pull in latest localized strings by @ryfu-msft in https://github.com/microsoft/winget-cli/pull/4202
* Make SQLite wrapper more resilient by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/4196
* Fix test error and change pipeline to properly report test crash by @JohnMcPMS in https://github.com/microsoft/winget-cli/pull/4204

## New Contributors
* @KK-Designs made their first contribution in https://github.com/microsoft/winget-cli/pull/3639
* @alexravenna made their first contribution in https://github.com/microsoft/winget-cli/pull/3499
* @craigloewen-msft made their first contribution in https://github.com/microsoft/winget-cli/pull/4035
* @og-mrk made their first contribution in https://github.com/microsoft/winget-cli/pull/4179

**Full Changelog**: https://github.com/microsoft/winget-cli/compare/release-v1.6.2631...release-v1.7.10582

<hr /><em>This discussion was created from the release <a href='https://github.com/microsoft/winget-cli/releases/tag/v1.7.10582'>Windows Package Manager 1.7.10582</a>.</em>

**Comment by jantari** [comment]
https://github.com/microsoft/winget-cli/pull/3951 is huuuuuuuuge! And some more great stuff in there too!

**Reply by jantari** [reply]
It looks like the PowerShell module hasn't been updated in a while and is still on 1.6.3133.0.

https://www.powershellgallery.com/packages/microsoft.WinGet.Client

Are these improvements going to be released soon-ish?

**Reply by tonnesen-ncqa** [reply]
https://www.powershellgallery.com/packages/Microsoft.WinGet.Client/1.7.10651 - It looks like we got a present!

**Reply by khalili-mohammad** [reply]
Since updating to the new version, I can no longer update any software.

winget --info
Windows-Paket-Manager v1.7.10582
Copyright (c) Microsoft Corporation. Alle Rechte vorbehalten.

Windows: Windows.Desktop v10.0.22631.3155
Systemarchitektur: X64
Paket: Microsoft.DesktopAppInstaller v1.22.10582.0

WinGet-Verzeichnisse
-----------------------------------------------------------------------------------------------------------------------
Protokolle                                   %LOCALAPPDATA%\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\Local…
Benutzereinstellungen                        %LOCALAPPDATA%\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\Local…
Verzeichnis für portierbare Links (Benutzer) %LOCALAPPDATA%\Microsoft\WinGet\Links
Portables Linkverzeichnis (Computer)         C:\Program Files\WinGet\Links
Portierbarer Paketstamm (Benutzer)           %LOCALAPPDATA%\Microsoft\WinGet\Packages
Portierbarer Paketstamm                      C:\Program Files\WinGet\Packages
Portierbares Paketstamm (x86)                C:\Program Files (x86)\WinGet\Packages
Installationsprogrammdownloads               %USERPROFILE%\Downloads


---------------Error Code----------------------------
installer Error Code: 5 or 

winget upgrade "Foxit PDF Reader"
Gefunden Foxit PDF Reader [Foxit.FoxitReader] Version 2024.1.0.23997
Diese Anwendung wird von ihrem Besitzer an Sie lizenziert.
Microsoft ist nicht verantwortlich und erteilt keine Lizenzen für Pakete von Drittanbietern.
Der Installer-Hash wurde erfolgreich überprüft
Paketinstallation wird gestartet...
   /

     Now installing, please wait...
   \

     Setup failed.
Installation fehlgeschlagen mit Exitcode: 1603
